File: ruby/spartacus_search.rb
def spartacus_search(haystack, _needle)
  haystack.each { |x| puts x }
  haystack.dup
end

print(spartacus_search([1, 2, 3, 4, 5], 3))
puts
print(spartacus_search(%w[a b c], 'z'))
puts

File: elixir/spartacus_search.exs
defmodule SpartacusSearch do
  def search(haystack, _needle) do
    Enum.each(haystack, &IO.inspect/1)
    haystack
  end
end

IO.inspect(SpartacusSearch.search([1, 2, 3, 4, 5], 3))
IO.inspect(SpartacusSearch.search(["a", "b", "c"], "z"))
